tests/unit_tests/sub_classing_test.py

```diff
@@ -0,0 +1,114 @@
+import unittest
+import numpy as np
+
+from super_gradients.training.datasets import DetectionDataSetV2
+from super_gradients.training.utils.detection_utils import DetectionTargetsFormat
+from super_gradients.training.exceptions.dataset_exceptions import EmptyDatasetException
+
+
+class TestDetectionDataSet(DetectionDataSetV2):
+    def __init__(self, input_dim, *args, **kwargs):
+        """Dummy Dataset testing subclassing, designed with no annotation that includes class_2."""
+
+        self.DUMMY_TARGETS = [np.array([[0, 0, 10, 10, 0],
+                                        [0, 5, 10, 15, 0],
+                                        [0, 5, 15, 20, 0]]),
+                              np.array([[0, 0, 10, 10, 0],
+                                        [0, 5, 10, 15, 0],
+                                        [0, 15, 55, 20, 1]])]
+
+        self.image_size = input_dim
+        kwargs['n_available_samples'] = len(self.DUMMY_TARGETS)
+        kwargs['all_classes_list'] = ["class_0", "class_1", "class_2"]
+        kwargs['original_target_format'] = DetectionTargetsFormat.XYXY_LABEL
+        super().__init__(input_dim=input_dim, *args, **kwargs)
+
+    def _load_annotation(self, sample_id: int) -> dict:
+        """Load 2 different annotations.
+            - Annotation 0 is made of: 3 targets of class 0, 0 of class_1 and 0 of class_2
+            - Annotation 1 is made of: 2 targets of class_0, 1 of class_1 and 0 of class_2
+        """
+        return {"img_path": "", "target": self.DUMMY_TARGETS[sample_id]}
+
+    # DetectionDatasetV2 will call _load_image but since we don't have any image we patch this method with
+    # tensor of image shape
+    def _load_image(self, index: int) -> np.ndarray:
+        return np.random.random(self.image_size)
+
+
+class TestDatasetInterface(unittest.TestCase):
+    def setUp(self) -> None:
+        self.CONFIG_KEEP_EMPTY_ANNOTATION = [
```

Comment:
Again, what's with all these uppercase letters ? Where did you see we do this ?

Reply:
Same, changed in d5bc0d4b6a901e2db163f8fa85597a762e1142eb